core/cat/mad_hatter/plugin.py

```diff
@@ -207,18 +207,18 @@ def _load_decorated_functions(self):
         plugin_overrides = []
 
         for py_file in self.py_files:
-            py_filename = py_file.replace("/", ".").replace(".py", "")  # this is UGLY I know. I'm sorry
+            module_name = os.path.splitext(os.path.basename(py_file))[0]
 
-            log.info(f"Import module {py_filename}")
+            log.info(f"Import module {py_file}")
 
             # save a reference to decorated functions
             try:
-                plugin_module = importlib.import_module(py_filename)
+                plugin_module = imp.load_source(module_name, py_file)
```

Comment:
why not importlib?

Reply:
I didn't find a clean way to do it with importlib. If you have a better way to do that, I accept suggestions 👍🏽

Reply:
Can you try something like this, please?
```python
spec=importlib.util.spec_from_file_location(module_name, py_file)
plugin_module = importlib.util.module_from_spec(spec)
```
I think `imp` package is deprecated

Reply:
I tried that way but it says it doesn't find any hooks

Reply:
@nicola-corbellini I found a way to use importlib, if you want to check it out